public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -61,7 +61,7 @@ const RulesFilter = () => {
       queryString: null,
       dataSource: null,
     });
-    setFilterKey(filterKey + 1);
+    setTimeout(() => setFilterKey(filterKey + 1), 100);
```

Comment:
This is to avoid a race condition between updating the query params and resetting the key. The key would be reset before the query params updated and would still get the previous value for the inputs `defaultValue`.